diff --git apachesolr_search_blocks.module apachesolr_search_blocks.module
index 6f08ec4..71e6330 100644
--- apachesolr_search_blocks.module
+++ apachesolr_search_blocks.module
@@ -104,7 +104,7 @@ function apachesolr_search_blocks_form_apachesolr_search_page_settings_form_alte
     // Load up the defaults from the database if they exist.
     $edit = db_select('apachesolr_search_blocks', 'a')
       ->fields('a')
-      ->condition('page_id', $form['search_page']['#value']->page_id)
+      ->condition('page_id', $form['search_page']['#value']['page_id'])
       ->execute()
       ->fetchAssoc();
   }
