diff --git a/apachesolr_access/apachesolr_access.module b/apachesolr_access/apachesolr_access.module
index bcb3557..d330870 100644
--- a/apachesolr_access/apachesolr_access.module
+++ b/apachesolr_access/apachesolr_access.module
@@ -60,11 +60,9 @@ function apachesolr_access_build_subquery($account) {
     throw new Exception('No access');
   }
   $node_access_query = apachesolr_drupal_subquery();
-  if (user_access('bypass node access', $account)) {
-    // Access all content from the current site.
-    $node_access_query->addFilter('hash', apachesolr_site_hash());
-  }
-  else {
+  $node_access_query->addFilter('hash', apachesolr_site_hash());
+  // If user doesn't have bypass node access set grants on specific nodes
+  if (user_access('bypass node access', $account) === FALSE) {
     // Get node access grants.
     $grants = node_access_grants('view', $account);
     foreach ($grants as $realm => $gids) {
